source/kraken/fill_disruption_from_chaos.cpp

```diff
@@ -310,37 +337,34 @@ struct apply_impacts_visitor : public boost::static_visitor<> {
 
 };
 
-struct functor_add_vj {
+struct functor_copy_vj_to_jp {
     const boost::shared_ptr<nt::new_disruption::Impact>& impact;
     nt::JourneyPattern* jp;
-    const nt::StopPoint* stop_point;
+    const std::set<const nt::StopPoint*>& impacted_stop_points;
     nt::PT_Data& pt_data;
     const nt::MetaData& meta;
 
-    functor_add_vj(const boost::shared_ptr<nt::new_disruption::Impact>& impact,
-            nt::JourneyPattern* jp, const nt::StopPoint* stop_point,
-            nt::PT_Data& pt_data, const nt::MetaData& meta) :
-        impact(impact), jp(jp), stop_point(stop_point), pt_data(pt_data), meta(meta) {}
-
     //return true only if the new vj circulate at least one day
-    bool init_vj(nt::VehicleJourney* vj, nt::VehicleJourney& vj_ref) const {
-        vj->journey_pattern = jp;
-        vj->idx = pt_data.vehicle_journeys.size();
-        vj->uri = make_adapted_uri_fast(vj_ref.uri, pt_data.vehicle_journeys.size());
-        vj->is_adapted = true;
-        vj->name = vj_ref.name;
-        vj->company = vj_ref.company;
+    bool init_vj(nt::VehicleJourney& vj, nt::VehicleJourney& vj_ref) const {
```

Comment:
* :+1: for & instead of *
* rename vj -> adapted_vj ?
* const& for vj_ref (but it's adapted_validity_pattern is modified... see remark below too)